src/player/player.cpp

```diff
@@ -122,38 +150,24 @@ CommandResult Player::Play()
 CommandResult Player::Quit()
 {
 	this->Eject();
-	this->state.Set(PlayerState::State::QUITTING);
-
-	// Quitting is always a valid command.
+	this->is_running = false;
 	return CommandResult::Success();
 }
 
 CommandResult Player::Seek(const std::string &time_str)
 {
-	if (!this->state.In(PlayerState::AUDIO_LOADED_STATES)) {
-		return CommandResult::Invalid(MSG_CMD_NEEDS_LOADED);
-	}
-
 	std::uint64_t pos = 0;
-	size_t cpos = 0;
+
 	try {
-		// In previous versions, this used to parse a unit at the end.
-		// This was removed for simplicity--use baps3-cli etc. instead.
-		pos = std::stoull(time_str, &cpos);
+		pos = SeekParse(time_str);
 	} catch (...) {
```

Comment:
should be changed to catch seek errors, and SeekParse should catch the std::exceptions, i think. also, this